src/main/java/uk/gov/di/ipv/cri/common/library/service/AccessTokenService.java

```diff
@@ -63,11 +67,18 @@ public AccessTokenResponse createToken(TokenRequest tokenRequest) {
         return new AccessTokenResponse(new Tokens(accessToken, null)).toSuccessResponse();
     }
 
-    public SessionItem updateSessionAccessToken(
+    public void updateSessionAccessToken(
             SessionItem sessionItem, AccessTokenResponse tokenResponse) {
+        // Set the access token
         sessionItem.setAccessToken(
                 tokenResponse.getTokens().getBearerAccessToken().toAuthorizationHeader());
-        return sessionItem;
+
+        // Set the access token expiry
+        sessionItem.setAccessTokenExpiryDate(
+                configurationService.getBearerAccessTokenExpirationEpoch());
+
+        // Expire the authorization code immediately, as it can only be used once
+        sessionItem.setAuthorizationCodeExpiryDate(clock.instant().getEpochSecond());
```

Comment:
would it be better to set the authorizationCode to null ? ( effectively revoke the authorizationCode )
Then the auth code expiry date can be left as is.
This might help with debugging in future...
for example, if an item exists in dynamodb with an auth code expiry date and no auth code then it would be evident that the auth code has been used 

It would also allow the dependency on Clock to be removed from this class